### PR TITLE
fix: address failure caused by method signature change in SPARK-48791

### LINKED ISSUE #692: address failure caused by method signature change in SPARK-48791
### Describe the bug

Comet `BatchReader` uses a private method in Spark `TaskMetrics.externalAccums`. In the latest codebase of Spark a fix for SPARK-48791 has changed the signature of this method which causes Comet to fail with 
```
org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 2.0 failed 1 times, most recent failure: Lost task 0.0 in stage 2.0 (TID 2) (localhost executor driver): java.lang.NoSuchMethodError: 'scala.collection.mutable.Buffer org.apache.spark.executor.TaskMetrics.externalAccums()'
```

### Steps to reproduce

Build Spark from latest code base and tru to run the `HiveParquetSuite` with Comet enabled.

### Expected behavior

The test should pass

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by parthchandra**
## Which issue does this PR close?

Closes #692 

## Rationale for this change

A private method in TaskMetrics that Comet accesses changed its signature

## What changes are included in this PR?

Use reflection to call the method

## How are these changes tested?

Local build of Spark with the breaking change and then running HiveParquetSuite

### CODE REVIEW

File: common/src/main/spark-3.3/org/apache/comet/shims/ShimBatchReader.scala [hunk 1]
```diff
@@ -33,4 +37,24 @@ object ShimBatchReader {
       Array.empty[String],
       0,
       0)
+
+  def getTaskAccumulator(taskMetrics: TaskMetrics): Option[AccumulatorV2[_, _]] = {
```
Comment (andygrove) [inline common/src/main/spark-3.3/org/apache/comet/shims/ShimBatchReader.scala:41]: This looks good, but it seems to be the same implementation in all of the shims? Deos it even need to be in the shim layer?

File: common/src/main/spark-3.3/org/apache/comet/shims/ShimBatchReader.scala [hunk 2]
```diff
@@ -19,8 +19,12 @@
...
 package org.apache.comet.shims
 
+import scala.collection.mutable
+
+import org.apache.spark.executor.TaskMetrics
 import org.apache.spark.sql.catalyst.InternalRow
 import org.apache.spark.sql.execution.datasources.PartitionedFile
+import org.apache.spark.util.AccumulatorV2
 
 object ShimBatchReader {
```
Comment (andygrove) [inline common/src/main/spark-3.3/org/apache/comet/shims/ShimBatchReader.scala:29]: nit: the shiimmed method is for getting an accumulator from TaskMetrics and does not seem specific to `BatchReader`, so maybe it should be in a specific `ShimTaskMetric` object?

File: common/src/main/spark-3.3/org/apache/comet/shims/ShimBatchReader.scala (hunk 1 above)
Comment (parthchandra) [reply common/src/main/spark-3.3/org/apache/comet/shims/ShimBatchReader.scala:41]: I could move this into the `BatchReader` itself. Perhaps that is cleaner.

File: common/src/main/spark-3.3/org/apache/comet/shims/ShimBatchReader.scala (hunk 2 above)
Comment (parthchandra) [reply common/src/main/spark-3.3/org/apache/comet/shims/ShimBatchReader.scala:29]: I believe we tend to put the methods in the class they are used. Anyway, let me move this to `BatchReader` and avoid the code duplication.

**Comment by parthchandra** [comment]
@andygrove removed the shims and moved the code into `BatchReader`.

**Review by viirya — APPROVED** [review]

**Review by andygrove — APPROVED** [review]

**Pull request merged**